sos/policies/distros/__init__.py

```diff
@@ -218,13 +218,12 @@ def _container_init(self):
         if ENV_CONTAINER in os.environ:
             if os.environ[ENV_CONTAINER] in ['docker', 'oci', 'podman']:
                 self._in_container = True
-        if ENV_HOST_SYSROOT in os.environ:
-            _host_sysroot = os.environ[ENV_HOST_SYSROOT]
-        use_sysroot = self._in_container and _host_sysroot is not None
-        if use_sysroot:
-            host_tmp_dir = os.path.abspath(_host_sysroot + self._tmp_dir)
-            self._tmp_dir = host_tmp_dir
-        return _host_sysroot if use_sysroot else None
+                if ENV_HOST_SYSROOT in os.environ:
+                    self._tmp_dir = os.path.abspath(
+                        os.environ[ENV_HOST_SYSROOT] + self._tmp_dir
```

Comment:
Can't be in some strange use case `os.environ[ENV_HOST_SYSROOT] = None`? The current code detects this, the new one doesnt.

Otherwise ACK and +1 to the simplification, this will be more readable.